module-2/backend/week_one.md

```diff
@@ -5,30 +5,44 @@ Fork this respository. Answer the questions to the best of your ability. Try to
+5. Why do we follow conventions when creating our actions/path names in our Sinatra routes? It makes it easier to follow for users and easier to comprehend for other developers.
+6. What types of variables are accessible in our view templates without explicitly passing them? Instance
 7. Given the following block of code, how would I pass an instance variable `count` with a value of `1` to my `index.erb` template?
   
   ```ruby
   get '/horses' do
-    erb :index
+    @count = 1
+    name = "Mr. Ed"
+    erb :index, locals: name
   end
   ```
 
 8. In the same code block, how would I pass a local variable `name` with a value of `Mr. Ed`?
-9. What's the purpose of ERB?
-10. Why do I need a development AND test database?
-11. What's responsive design?
-12. What is CRUD and why is it important?
-13. What does HTTP stand for? 
-14. What are the two ways to interpolate Ruby in an ERB view template? What's the difference between these two ways?
-15. What's an ORM?
-16. What's the most commonly used ORM?
+9. What's the purpose of ERB? It allows the HTML structure of the page to understand and interpolate Ruby code
+10. Why do I need a development AND test database? It allows the developer to only load as much data as necessary in the tests (improving speed) and keeps the real database free of irrelevant data the developer might add just for the purpose of testing.
+11. What's responsive design? The page's content responds to the window size.
+12. What is CRUD and why is it important? Create Read Update Destroy--it's the basic for framework necessary for any effective database
+13. What does HTTP stand for? Hypertext Transfer Protocol
+14. What are the two ways to interpolate Ruby in an ERB view template? What's the difference between these two ways? <% %> || <%= %> ; The latter specifies a return value, often but not always printed to the screen.
+15. What's an ORM? Object Relational Method
+16. What's the most commonly used ORM? ActiveRecord
 17. Let's say we have an application with restaurants. There are seven verb + path combinations necessary to provide full CRUD functionality for our restaurant application. List each of the seven combinations, and explain what each is for.
-18. What's a migration? 
-19. When you create a migration, does it automatically modify your database?
-20. How does a model relate to a database?
-21. What's the difference between agile workflow and waterfall method?
-22. What is the difference between `#new` and `#create`?
+get '/'
+get '/restaurants'
+get '/restaurants/new'
+post '/restaurants'
+get '/restaurants/:id'
+put '/restaurants/:id'
+delete '/restaurants/:id'
+18. What's a migration? Migrations create or update databases through the ORM.
+19. When you create a migration, does it automatically modify your database? Not necessarily, it just enacts your code if you did modify it.

```

Comment:
Would want to hear more about what you mean here. In general, a migration doesn't alter your database until you run `rake db:migrate`.